src/System.Private.CoreLib/src/Internal/Runtime/InteropServices/WindowsRuntime/InterfaceMarshalerSupport.cs

```diff
@@ -0,0 +1,13 @@
+using System;
+using System.StubHelpers;
+
+namespace Internal.Runtime.InteropServices.WindowsRuntime
+{
+    public static class InterfaceMarshalerSupport
```

Comment:
Do we need to have separate types for all these? I think all these support methods can be on a single type.

Reply:
Then the method name will be a little bit longer, such as InterfaceMarshaler_ConvertToManagedWithoutUnboxing, it is ok?

Reply:
The existing public APIs similar to this are called like `Marshal.GetObjectForIUnknown`, `Marshal.GetObjectForIUnknown`  or `Marshal.GetTypedObjectForIUnknown`. 

Can we follow the same naming convention here and call it `GetObjectForIUnknown...` ?

Also, is box/unbox the right name here? Unboxing has a specific meaning in .NET. As far as I can tell all other existing (public) APIs use wrap/unwrap in this context.

Reply:
>  Also, is box/unbox the right name here? Unboxing has a specific meaning in .NET.

@jkotas , This is related to WinRT unboxing. For function name, Do you prefer GetObjectForIUnknownNoWinRTUnboxing or GetObjectForIUnknownNoUnwrap?
I would prefer the 1st one.

Reply:
The use case/scenario seems to be related to WinRT, but the underlying function implementation doesn't appear to be specific to WinRT - `stubhelpers.cpp` in 868. I see the `ObjFromComIP::IGNORE_WINRT_AND_SKIP_UNBOXING` but that indicates the WinRT aspect should be ignored, so what happens if we pass in a non-WinRT object? I agree with @jkotas here and think something like `GetObjectForIUnknownNoUnboxing` seems to be exactly what the function does and documenting an example of usage should alleviate any confusion - "Usually used for WinRT scenarios..."

Reply:
This is only used in two places in CoreFX: once for `INotifyCollectionChangedEventArgs` and second time for `IPropertyChangedEventArgs`.

Why does the WinRT unboxing matters for these two places? Can't this just use the vanilla `Marshal.GetUniqueObjectForIUnknown` ?

If this really needs to be special cases for WinRT unboxing, my name preference would be `GetObjectForIUnknownWithoutUnboxing`.

Reply:
> This is only used in two places in CoreFX: once for INotifyCollectionChangedEventArgs and second time for IPropertyChangedEventArgs.

There are two more places: https://github.com/dotnet/coreclr/blob/f31097f14560b193e76a7b2e1e61af9870b5356b/src/System.Private.CoreLib/src/System/StubHelpers.cs#L1295
https://github.com/dotnet/coreclr/blob/f31097f14560b193e76a7b2e1e61af9870b5356b/src/System.Private.CoreLib/src/System/StubHelpers.cs#L1461

>Why does the WinRT unboxing matters for these two places? Can't this just use the vanilla Marshal.GetUniqueObjectForIUnknown ?

For functional(correctness) point view, actually it doesn't matter. I suppose the reason why we use unboxing one is convenient and a little bit perf gain. Let's use Marshal.GetUniqueObjectForIUnknown instead

Reply:
> There are two more places:

These are internal to CoreLib. They do not need the public API.

> Let's use Marshal.GetUniqueObjectForIUnknown instead

Agree.

Reply:
Remove these *support.cs